lib/connectors/memory.js

```diff
@@ -148,11 +148,13 @@ Memory.prototype.saveToFile = function (result, callback) {
   if (file) {
     if(!self.writeQueue) {
       // Create a queue for writes
+      var cache = self.cache;
+      var ids = self.ids;
       self.writeQueue = async.queue(function (task, cb) {
         // Flush out the models/ids
         var data = JSON.stringify({
-          ids: self.ids,
-          models: self.cache
+          ids: ids,
+          models: cache
```

Comment:
If I use the following block of code, self.ids and self.cache does not reflect the latest cache and ids (It misses the last item)

```
        var data = JSON.stringify({
          ids: self.ids,
          models: self.cache
        }, null, '  ');
```
However the following code works; cache and ids reflect the updated cache and ids:

```
      var cache = self.cache;
      var ids = self.ids;
      self.writeQueue = async.queue(function (task, cb) {
        // Flush out the models/ids
        var data = JSON.stringify({
          ids: ids,
          models: cache
        }, null, '  ');

        fs.writeFile(self.settings.file, data, function (err) {
          cb(err);
          task.callback && task.callback(err, task.data);
        });
      }, 1);
```